python/pyspark/sql/pandas/serializers.py

```diff
@@ -2141,6 +2188,11 @@ def generate_data_batches(batches):
 
             def extract_rows(cur_batch, col_name, key_offsets):
                 data_column = cur_batch.column(cur_batch.schema.get_field_index(col_name))
+
+                # Check if the entire column is null
+                if data_column.null_count == len(data_column):
+                    return None
```

Comment:
Given we've changed the implicit type signature of the function lets maybe add a type annotation on generate_data_batches for readability.

Reply:
Done